raiden/network/rpc/client.py

```diff
@@ -161,6 +163,25 @@ class JSONRPCClient(object):
         nonce_offset (int): Network's default base nonce number.
     """
 
+    def _checkNodeConnection(func, *args, **kwargs):
+
+        def retry_if_disconnect(self, *args, **kwargs):
+            for i, timeout in enumerate(timeout_two_stage(10, 3, 10)):
+                try:
+                    result = func(self, *args, **kwargs)
+                    if i > 0:
+                        log.info('Client reconnected')
+                    return result
+
+                except (requests.exceptions.ConnectionError, InvalidReplyError):
```

Comment:
Why is the client considered offline when an `InvalidReplyError` is raised?

I'm wondering what is the content of the response, perhaps the node restarted and Raiden polled for a filter that is gone?

Reply:
Just after the disconnect `tinyrpc` gets an empty response which leads to the `InvalidReplyError` and in my testing only happens in this case.

Not sure why this happens.